### ISSUE TITLE
dirduster crashes on start (ubuntu22)

### ISSUE THREAD

**Issue by Trolldemorted** [cym13/dirduster#8]
In a fresh ubuntu22 container (`docker run --rm -it ubuntu`), dirduster fails with the following stack trace:
```
-- Scanning https://[redacted] --

object.Exception@../../../.dub/packages/requests-1.0.8/requests/source/requests/ssl_adapter.d(249): can't complete call to TLS_method
----------------
??:? const requests.ssl_adapter.SSL_METHOD* requests.ssl_adapter.OpenSSL.TLS_method() [0x5596c7160228]
??:? void requests.streams.OpenSslSocket.initSsl(requests.streams.SSLOptions) [0x5596c71432b2]
??:? requests.streams.OpenSslSocket requests.streams.OpenSslSocket.__ctor(std.socket.AddressFamily, std.socket.SocketType, requests.streams.SSLOptions) [0x5596c7143878]
??:? void requests.streams.SSLSocketStream.open(std.socket.AddressFamily) [0x5596c7143c27]
??:? requests.streams.SocketStream requests.streams.SocketStream.connect(immutable(char)[], ushort, core.time.Duration) [0x5596c7144310]
??:? requests.streams.SocketStream requests.streams.SSLSocketStream.connect(immutable(char)[], ushort, core.time.Duration) [0x5596c7143d06]
??:? requests.streams.NetworkStream requests.http.HTTPRequest.setupConnection() [0x5596c7156f76]
??:? requests.http.HTTPResponse requests.http.HTTPRequest.exec_from_parameters() [0x5596c715ba72]
??:? requests.http.HTTPResponse requests.http.HTTPRequest.execute(requests.request.Request) [0x5596c715d5ee]
??:? requests.base.Response requests.request.Request.LastInterceptor.opCall(requests.request.Request, requests.request.RequestHandler) [0x5596c712f343]
??:? requests.base.Response requests.request.RequestHandler.handle(requests.request.Request) [0x5596c712e3d3]
??:? requests.base.Response requests.request.Request.execute(immutable(char)[], immutable(char)[], std.typecons.Tuple!(immutable(char)[], "key", immutable(char)[], "value").Tuple[]) [0x5596c712feef]
??:? int app.scanUrl(immutable(char)[], const(immutable(char)[])[], requests.request.Request[], ushort[], immutable(char)[][immutable(char)[]], std.regex.internal.ir.Regex!(char).Regex, immutable(char)[]).__foreachbody8(ulong, ref requests.request.Request) [0x5596c70c6a1b]
??:? void std.parallelism.ParallelForeach!(requests.request.Request[]).ParallelForeach.opApply(scope int delegate(ulong, ref requests.request.Request)).doIt() [0x5596c70cf99f]
??:? void std.parallelism.run!(void delegate()).run(void delegate()) [0x5596c71abbdf]
??:? void std.parallelism.Task!(std.parallelism.run, void delegate()).Task.impl(void*) [0x5596c71ab6b7]
??:? void std.parallelism.AbstractTask.job() [0x5596c71aa14e]
??:? void std.parallelism.submitAndExecute(std.parallelism.TaskPool, void delegate()) [0x5596c71ab2c6]
??:? int std.parallelism.ParallelForeach!(requests.request.Request[]).ParallelForeach.opApply(scope int delegate(ulong, ref requests.request.Request)) [0x5596c70cf886]
??:? immutable(char)[][] app.scanUrl(immutable(char)[], const(immutable(char)[])[], requests.request.Request[], ushort[], immutable(char)[][immutable(char)[]], std.regex.internal.ir.Regex!(char).Regex, immutable(char)[]) [0x5596c70c6844]
??:? _Dmain [0x5596c70c7b6a]
```

after `apt install libssl-dev`, dirduster fails with the following message:
```
object.Exception@../../../.dub/packages/requests-1.0.8/requests/source/requests/ssl_adapter.d(139): loading openssl: unknown version for init
```

**Comment by cym13** [comment]
Hi, thanks for this issue. Unfortunately I couldn't reproduce it. Here's what I did:

    sudo docker run --rm -it ubuntu # as suggested

Then in docker:

    apt update
    apt install dub git build-essential
    git clone https://github.com/cym13/dirduster
    cd dirduster
    dub build
    ./dirduster -f ./DirBuster-Lists/directory-list-2.3-small.txt "$testwebsite"
    # Proper results, no crash
    ./dirduster --version
    # 1.7.0

Could you check again on your end maybe? Maybe your docker's network stack is configured in a special way? I'm at a loss to help you here.

EDIT: also I'd be curious to know whether you can reproduce that against other websites, maybe this is website specific such as being unable to agree on a TLS ciphersuite?

**Comment by Trolldemorted** [comment]
Thanks for the swift response! I was using your binary releases, and haven't tried building it yet. I just tested again (`docker run --rm -it ubuntu bash`):

```
apt update && apt install wget
cd ~
wget https://github.com/cym13/dirduster/releases/download/1.5.0/dirduster_1.5.0_linux_x86_64
chmod +x dirduster_1.5.0_linux_x86_64
echo "foo" > test.txt
./dirduster_1.5.0_linux_x86_64 -f test.txt https://google.com # fails with stacktrace
apt install libssl-dev
./dirduster_1.5.0_linux_x86_64 -f test.txt https://google.com # fails with "object.Exception@../../../.dub/packages/requests-1.0.8/requests/source/requests/ssl_adapter.d(139): loading openssl: unknown version for init"
```

**Comment by cym13** [comment]
I see, it's true that I hadn't created any release for the latest versions. I just published a release for the 1.7.1 (new minor as of today to switch to the latest branch of the requests library). Can you try again with that version and confirm any issue?

**Comment by Trolldemorted** [comment]
works like a charm, thanks a lot!
